web.py

```diff
@@ -466,31 +480,41 @@ def runloop():
             logger.info("WebGreenlet webserver starting (addr: %s, port: %d)", self.addr, self.port)
             socketio.run(app, host=self.addr, port=self.port)
 
-        def process_proposals_loop():
+        def process_periodic_events_loop():
+            current = int(time.time())
+            proposals_timer_last = current
+            email_alerts_timer_last = current
             while True:
-                gevent.spawn(process_proposals)
-                gevent.sleep(30)
+                current = time.time()
+                if current - proposals_timer_last > 30:
+                    gevent.spawn(process_proposals)
+                    proposals_timer_last += 30
+                if current - email_alerts_timer_last > 1800:
+                    gevent.spawn(process_email_alerts)
+                    email_alerts_timer_last += 1800
+                gevent.sleep(5)
 
         def start_greenlets():
             if SERVER_MODE == SERVER_MODE_WAVES:
                 logger.info("checking wallet...")
                 self.check_wallet()
             logger.info("starting WebGreenlet runloop...")
             self.runloop_greenlet.start()
-            self.process_proposals_greenlet.start()
+            self.process_periodic_events_greenlet.start()
 
         # create greenlet
         self.runloop_greenlet = gevent.Greenlet(runloop)
-        self.process_proposals_greenlet = gevent.Greenlet(process_proposals_loop)
+        # pylint: disable=attribute-defined-outside-init
```

Comment:
dont put this here, fix the error, change var `process_proposals_greenlet` in `__init__()`

Reply:
added to init and deleted the old ones not used anymore